chore: optimize rpc response time and error description

ashuralyk opened the pull request:
Optimize with the following 6 changes:
1. cache block receipts to eliminate the recent request cost of calling `get_block_receipts()` rpc within 512 slots
2. remove 3 times of calling `fetch_live_cells()` rpc
3. report an error to replace returning `None` when calling the `forcerelay_getCkbForcerelayTransaction` method
4. use a peaceful method to handle the insecure exit progress, e.g. ctrl-c pressed or internal panic
5. enable a safe exit while in the progress of chasing the tip
6. enable the `forcerelay_getCkbForcerelayTransaction` rpc response while in the chasing progress, and the response is showing the verifier is in CHASING status

Commit:
chore: optimize rpc response time and error description

Commit:
feat: use a peaceful method to process the abnormal exit

ashuralyk commented:
@ImJeremyHe 
![image](https://user-images.githubusercontent.com/3871265/224520864-2c27efbb-84ce-43a0-bc5e-7908640b76a1.png)
please help me check this error cause I changed the result type of `forcerelay_getCkbForcerelayTransaction`

ImJeremyHe commented:
> @ImJeremyHe ![image](https://user-images.githubusercontent.com/3871265/224520864-2c27efbb-84ce-43a0-bc5e-7908640b76a1.png) please help me check this error cause I changed the result type of `forcerelay_getCkbForcerelayTransaction`

It seams that the verifier was not started or that it was started but was not ready for receiving requests

ImJeremyHe committed:
fix: integration test

Commit:
bug: solve rpc server drop failed problem

chore: divide assemble_partial_verification_transaction function into three pieces

Commit:
bug: fix test_fetch_latest_checkpoints error caused by the unstable RPC in real world
